src/main.ts

```diff
@@ -39,42 +42,75 @@ export default function transform({
-    const store: Array<StoreItem> = [];
+  const store: Array<StoreItem> = [];
 
-    const ethTx = toEthTx({ transaction, header, receipt });
-    // Can be null if:
-    // 1. The transaction is missing calldata.
-    // 2. The transaction is a multi-call.
-    // 3. The length of the signature array is different from 5.
-    // 4. The chain id is not encoded in the v param of the signature for a
-    //    Legacy transaction.
-    // 5. The deserialization of the transaction fails.
-    if (ethTx === null) {
-      return [];
-    }
-    store.push({ collection: "transactions", data: { tx: ethTx } });
+  await Promise.all(
+    (events ?? []).map(async ({ transaction, receipt, event }, i) => {
+      const typedEthTx = toTypedEthTx({ transaction });
+      // Can be null if:
+      // 1. The transaction is missing calldata.
+      // 2. The transaction is a multi-call.
+      // 3. The length of the signature array is different from 5.
+      // 4. The chain id is not encoded in the v param of the signature for a
+      //    Legacy transaction.
+      // 5. The deserialization of the transaction fails.
+      if (typedEthTx === null) {
+        return null;
+      }
+      const ethTx = toEthTx({ transaction: typedEthTx, header, receipt });
+      // Can be null if:
+      // 1. The typed transaction if missing a signature param (v, r, s).
+      if (ethTx === null) {
+        return null;
+      }
 
-    // Can be null if:
-    // 1. The event is part of the defined ignored events (see IGNORED_KEYS).
-    // 2. The event has an invalid number of keys.
-    const ethLogs = receipt.events.map((e) => {
-      return toEthLog({ transaction: ethTx, event: e });
-    }).filter((e) => e !== null) as JsonRpcLog[];
-    ethLogs.forEach((ethLog) => {
-      store.push({ collection: "logs", data: { log: ethLog } });
-    });
+      // Can be null if:
+      // 1. The event is part of the defined ignored events (see IGNORED_KEYS).
+      // 2. The event has an invalid number of keys.
+      const ethLogs = receipt.events.map((e) => {
+        return toEthLog({ transaction: ethTx, event: e });
+      }).filter((e) => e !== null) as JsonRpcLog[];
 
-    const ethReceipt = toEthReceipt({
-      transaction: ethTx,
-      logs: ethLogs,
-      event,
-      cumulativeGasUsed,
-    });
-    store.push({ collection: "receipts", data: { receipt: ethReceipt } });
-    cumulativeGasUsed += BigInt(ethReceipt.gasUsed);
+      const ethReceipt = toEthReceipt({
+        transaction: ethTx,
+        logs: ethLogs,
+        event,
+        cumulativeGasUsed,
+      });
 
-    return store;
+      // Add the transaction to the transaction trie.
+      await transactionTrie.put(RLP.encode(i), typedEthTx.serialize());
+      // Add the receipt to the receipt trie.
+      const encodedReceipt = encodeReceipt(
+        fromJsonRpcReceipt(ethReceipt),
+        typedEthTx.type,
+      );
+      await receiptTrie.put(RLP.encode(i), encodedReceipt);
```

Comment:
how can we test this? or have reference to the methodology?

Reply:
the tries come from:
- https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/block/src/block.ts#L85
- https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/vm/src/buildBlock.ts#L153

Reply:
let's add these links as comments to help documentation